scripts/library.sh

```diff
@@ -448,8 +442,34 @@ function get_canonical_path() {
   echo "$(cd ${path%/*} && echo $PWD/${path##*/})"
 }
 
+# Return the version for the current branch.
+function get_branch_version() {
+  local branch_name="$(git rev-parse --abbrev-ref HEAD)"
+  # Return the actual version number for the release branch.
+  # For example, we will return "0.5" if the branch name is "release-0.5".
+  if [[ $name =~ ^release-[0-9\.]+ ]]; then
+    echo ${branch_name} | sed 's/release-//g'
+  # Return empty string for any other branch.
+  else
+    echo ""
+  fi
+}
+
 # Initializations that depend on previous functions.
 # These MUST come last.
 
 readonly _TEST_INFRA_SCRIPTS_DIR="$(dirname $(get_canonical_path ${BASH_SOURCE[0]}))"
 readonly REPO_NAME_FORMATTED="Knative $(capitalize ${REPO_NAME//-/})"
+
+# Public latest stable nightly images and yaml files.
+readonly BRANCH_VERSION="$(get_branch_version)"
```

Comment:
Thanks, this is way nicer. Once last simplification: you don't need `BRANCH_VERSION` and top-level `if`. `get_branch_version()` can already return the right string to use as `KNATIVE_BASE_YAML_SOURCE`; it even makes sense to rename it to something like `get_knative_base_yaml_source()`.

Reply:
Done.